Orange/base.py

```diff
@@ -18,13 +19,12 @@ class Learner:
     name = 'learner'
     #: A sequence of data preprocessors to apply on data prior to
     #: fitting the model
-    preprocessors = ()
+    preprocessors = None
     learner_adequacy_err_msg = ''
 
     def __init__(self, preprocessors=None):
-        if preprocessors is None:
-            preprocessors = type(self).preprocessors
-        self.preprocessors = list(preprocessors)
+        self.use_default_preprocessors = False
```

Comment:
I disagree with everything except that it deserves a comment to explain why all this is necessary -- which would also answer your considerations about unnecessarily complicating things, about mandatory preprocessors and about `use_default_processors` (which is not a property of the class that should be given as an argument but rather a flag that would essentially need to be given in the `__call__` - except that it can't).

@pavlin-policar explained the workings of this two days ago and I already forgot. Better documentation would help an eager pull request reviewer understand the pull request better and thus give more informed comments. :)